internal/function/is_remote.go

```diff
@@ -0,0 +1,55 @@
+package function
+
+import (
+	"reflect"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+)
+
+// IsRemote checks the given string is a tag name.
+type IsRemote struct {
+	expression.UnaryExpression
+}
+
+// NewIsRemote creates a new IsRemote function.
+func NewIsRemote(e sql.Expression) *IsRemote {
+	return &IsRemote{expression.UnaryExpression{Child: e}}
+}
+
+var _ sql.Expression = (*IsRemote)(nil)
```

Comment:
What's the use of this line?

Reply:
check `IsRemote` implements `sql.Expression` on compile-time

Reply:
Thanks, understood. :+1:

Reply:
it feels weird that a `expression.UnaryExpression` doesn't implement the `Expression` interface :confused:

Reply:
Think of it as an "abstract class" that expressions extend from. There are some things `UnaryExpression` cannot implement, such as `Type` and `Name`, because it's meant for you to implement it on the concrete expression. And there are some others like `TransformUp` that it could implement but that could lead you to forget redefining it in your concrete expression.